internal/drivers/ankabuild/client.go

```diff
@@ -0,0 +1,206 @@
+	ScriptMonitoring       bool   `json:"script_monitoring,omitempty"`
+	ScriptTimeout          int    `json:"script_timeout,omitempty"`
+	ScriptFailHandler      int    `json:"script_fail_handler,omitempty"`
+	VMID                   string `json:"vmid,omitempty"`
+	GroupID                string `json:"group_id,omitempty"`
+	Priority               string `json:"priority,omitempty"`
+	Tag                    string `json:"tag,omitempty"`
+}
+
+type vmResponse struct {
+	Status  string `json:"status"`
+	Message string `json:"message"`
+	Body    struct {
+		InstanceID    string `json:"instance_id"`
+		InstanceState string `json:"instance_state"`
+		AnkaRegistry  string `json:"anka_registry"`
+		Vmid          string `json:"vmid"`
+		Tag           string `json:"tag"`
+		Vminfo        struct {
+			UUID                string    `json:"uuid"`
+			Name                string    `json:"name"`
+			CPUCores            int       `json:"cpu_cores"`
+			RAM                 string    `json:"ram"`
+			Status              string    `json:"status"`
+			NodeID              string    `json:"node_id"`
+			HostIP              string    `json:"host_ip"`
+			IP                  string    `json:"ip"`
+			VncPort             int       `json:"vnc_port"`
+			VncConnectionString string    `json:"vnc_connection_string"`
+			VncPassword         string    `json:"vnc_password"`
+			CreationDate        time.Time `json:"creation_date"`
+			StopDate            time.Time `json:"stop_date"`
+			Version             string    `json:"version"`
+			PortForwarding      []struct {
+				GuestPort int    `json:"guest_port"`
+				HostPort  int    `json:"host_port"`
+				Protocol  string `json:"protocol"`
+				RuleName  string `json:"rule_name"`
+			} `json:"port_forwarding,omitempty"`
+		} `json:"vminfo"`
+		NodeID   string    `json:"node_id"`
+		TS       time.Time `json:"ts"`
+		CrTime   time.Time `json:"cr_time"`
+		Progress int       `json:"progress"`
+		Arch     string    `json:"arch"`
+		Vlan     string    `json:"vlan"`
+	} `json:"body"`
+}
+
+type statusResponse struct {
+	Status  string `json:"status"`
+	Message string `json:"message"`
+	Body    struct {
+		Status          string `json:"status"`
+		Version         string `json:"version"`
+		RegistryAddress string `json:"registry_address"`
+		RegistryStatus  string `json:"registry_status"`
+		License         string `json:"license"`
+	} `json:"body"`
+}
+
+type createVMResponse struct {
+	Status  string   `json:"status"`
+	Body    []string `json:"body"`
+	Message string   `json:"message"`
+}
+
+type deleteVMRequest struct {
+	ID string `json:"id"`
+}
+
+type client struct {
+	client    *http.Client
+	addr      string
+	authToken string
+}
+
+type Client interface {
+	VMCreate(ctx context.Context, params *createVMParams) (*createVMResponse, error)
+	VMDelete(ctx context.Context, id string) error
+	VMFind(ctx context.Context, id string) (*vmResponse, error)
+	Status(ctx context.Context) (*statusResponse, error)
+}
```

Comment:
why do we have an interface ? it is only implemented once
unless you are adding tests based on the interface

Reply:
I'll leave because we should extend out unit tests when we've got the time